packages/angular/src/builders/webpack-server/lib/configure-buildable-libs-support.ts

```diff
@@ -0,0 +1,57 @@
+import { BuilderContext } from '@angular-devkit/architect';
+import { executeDevServerBuilder } from '@angular-devkit/build-angular';
+import { AngularWebpackPlugin } from '@ngtools/webpack';
+import {
+  joinPathFragments,
+  readCachedProjectGraph,
+  TargetConfiguration,
+} from '@nrwl/devkit';
+import { WebpackNxBuildCoordinationPlugin } from '@nrwl/web/src/plugins/webpack-nx-build-coordination-plugin';
+import {
+  calculateProjectDependencies,
+  createTmpTsConfig,
+} from '@nrwl/workspace/src/utilities/buildable-libs-utils';
+import { isNpmProject } from 'nx/src/project-graph/operators';
+
+type Transforms = Parameters<typeof executeDevServerBuilder>[2];
+type WebpackConfiguration = Parameters<Transforms['webpackConfiguration']>[0];
+
+export function configureBuildableLibsSupport(
+  baseWebpackConfig: WebpackConfiguration,
+  context: BuilderContext,
+  buildTargetName: string,
+  buildTarget: TargetConfiguration
+) {
+  const { target, dependencies } = calculateProjectDependencies(
+    readCachedProjectGraph(),
+    context.workspaceRoot,
+    context.target.project,
+    buildTargetName,
+    context.target.configuration
+  );
+  const tsConfig = createTmpTsConfig(
+    joinPathFragments(context.workspaceRoot, buildTarget.options.tsConfig),
+    context.workspaceRoot,
+    target.data.root,
+    dependencies
+  );
+
+  // We can't just pass the tsconfig path in memory to the angular builder
+  // function because we can't pass the build target options to it, the build
+  // targets options will be retrieved from the project configuration.
+  // Therefore, we overwrite the tsconfig path that comes in the webpack
+  // configuration for the AngularWebpackPlugin to use the temporary tsconfig.
+  const angularWebpackPlugin = baseWebpackConfig.plugins.find(
+    (plugin) => plugin.constructor.name === AngularWebpackPlugin.name
+  ) as unknown as AngularWebpackPlugin;
```

Comment:
This double cast is interesting. What's the reasoning behind this?

Reply:
The `baseWebpackConfig.plugins` is of type `WebpackPluginInstance` which can't be casted directly to `AngularWebpackPlugin`.